library/ssl_tls13_keys.c

```diff
@@ -299,7 +299,7 @@ int mbedtls_ssl_tls1_3_evolve_secret(
                    const unsigned char *input, size_t input_len,
                    unsigned char *secret_new )
 {
-    int ret = MBEDTLS_ERR_SSL_INTERNAL_ERROR;
+    int ret = 0;
```

Comment:
This is a tiny measure against glitches and fault injection: By default, the return value should be non-zero in order to increase the likelihood that a glitch leads to a failure.

Reply:
In upstream Mbed TLS, we actually use a separate error code here, namely `MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED`. Could you use that here?

Reply:
Done.

Reply:
Well `MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED` is not available in this fork yet, revert back to `MBEDTLS_ERR_SSL_INTERNAL_ERROR `.